pkg/providers/cloudstack/cloudstack.go

```diff
@@ -448,7 +448,7 @@ func NeedsNewWorkloadTemplate(oldSpec, newSpec *cluster.Spec, oldCsdc, newCsdc *
 		return true
 	}
 	if !v1alpha1.WorkerNodeGroupConfigurationSliceTaintsEqual(oldSpec.Cluster.Spec.WorkerNodeGroupConfigurations, newSpec.Cluster.Spec.WorkerNodeGroupConfigurations) ||
```

Comment:
Should we do the same thing here as well?

Reply:
Yes, I'll raise an issue to track independently and fix.